File: blood-bank-be/bloodBank/src/main/java/ftn/uns/ac/rs/bloodbank/blood/model/BloodDonation.java
package ftn.uns.ac.rs.bloodbank.blood.model;

import lombok.*;

import javax.persistence.*;
import java.util.UUID;

@Entity(name = "BloodDonation")
@Table(name = "blood_donation")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BloodDonation {
    @Id
    @GeneratedValue
    @Column(name = "id",nullable = false,updatable = false,columnDefinition = "uuid")
    private UUID id;
    private Integer bloodUnit;
    @Enumerated(EnumType.STRING)
    private BloodType  bloodType;
    private String noteForDoctor;
}

File: blood-bank-fe/src/app/model/examination/BloodDonation.ts
export class BloodDonation {
  bloodType?: string;
  noteForDoctor?: string;
  bloodUnit?: number;

  constructor(bloodType: string, noteForDoctor: string, bloodUnit: number) {
    this.bloodType = bloodType;
    this.noteForDoctor = noteForDoctor;
    this.bloodUnit = bloodUnit;
  }
}
